markups/commonmark.py

```diff
@@ -0,0 +1,86 @@
+# vim: ts=8:sts=8:sw=8:noexpandtab
+
+# This file is part of python-markups module
+# License: BSD
+# Copyright: (C) Shengjing Zhu, 2016
+
+import markups.common as common
+from markups.abstract import AbstractMarkup, ConvertedMarkup
+
+try:
+	from CommonMark import HtmlRenderer, Parser
+	from CommonMark.common import escape_xml
+except:
+	pass
+
+
+class CommonMarkMarkup(AbstractMarkup):
+	"""Markup class for CommonMark language.
+	Inherits :class:`~markups.abstract.AbstractMarkup`.
+	"""
+	name = 'CommonMark'
+	attributes = {
+		common.LANGUAGE_HOME_PAGE: 'http://commonmark.org/',
+		common.MODULE_HOME_PAGE: 'https://github.com/rtfd/CommonMark-py/',
+		common.SYNTAX_DOCUMENTATION: 'http://spec.commonmark.org/'
+	}
+
+	# Note: the same as MarkdownMarkup
+	# file_extensions = ('.md', '.mkd', '.mkdn', '.mdwn', '.mdown', '.markdown')
+	# default_extension = '.md'
+
+	@staticmethod
+	def available():
+		try:
+			import CommonMark
+		except ImportError:
+			return False
+		return True
+
+	def __init__(self, filename=None, options={}):
+		self.enable_highlight = options.get('highlight', True)
```

Comment:
If there is only one option, can you remove the `options` keyword and directly list `enable_highlight=True` as the argument?